### PR TITLE
Dev/login screen

### PR DESCRIPTION
**Pull request by DurtDiv3r**
Add Inputs to Components.kt and remove InputField composable. Add initial Login screen layout and ViewModel.

### CODE REVIEW

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.materialcomposereader.R
+
+@Composable
+fun UsernameInputField(
```
Comment (drawers) [inline app/src/main/java/com/example/materialcomposereader/components/Components.kt:26]: Question:
Is this an input field for just the Login Screen?

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    modifier: Modifier = Modifier,
```
Comment (drawers) [inline app/src/main/java/com/example/materialcomposereader/components/Components.kt:27]: Looks like this modifier is not being used?

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    OutlinedTextField(
+        value = value,
+        onValueChange = {
+            onNewValue(it)
+        },
+        label = {
+            Text(text = stringResource(id = R.string.login_username))
+        },
+        singleLine = true,
+        textStyle = TextStyle(fontSize = 16.sp, color = MaterialTheme.colorScheme.onBackground),
```
Comment (drawers) [inline app/src/main/java/com/example/materialcomposereader/components/Components.kt:41]: Are there plans to use Typography from the theme?

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        keyboardOptions = KeyboardOptions(
+            keyboardType = KeyboardType.Text,
+            imeAction = ImeAction.Next,
+        ),
+    )
+}
+
+@Composable
+fun EmailInputField(
+    modifier: Modifier = Modifier,
```
Comment (drawers) [inline app/src/main/java/com/example/materialcomposereader/components/Components.kt:53]: Question:
Is this the correct order of parameters for a Composable? 

https://twitter.github.io/compose-rules/rules/#modifiers-should-have-default-parameters

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    )
+}
+
+@Composable
+fun PasswordInputField(
+    modifier: Modifier,
+    value: String,
+    onNewValue: (String) -> Unit,
+    enabled: Boolean,
+    passwordVisibility: MutableState<Boolean>,
```
Comment (drawers) [inline app/src/main/java/com/example/materialcomposereader/components/Components.kt:83]: Question:
Why is this parameter typed as `MutableState<Boolean>`?

File: app/src/main/java/com/example/materialcomposereader/screens/login/LoginScreen.kt [hunk 6]
```diff
@@ -1,10 +1,57 @@
...
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
 import androidx.navigation.NavController
 
 @Composable
 fun LoginScreen(navController: NavController) {
```
Comment (drawers) [inline app/src/main/java/com/example/materialcomposereader/screens/login/LoginScreen.kt:19]: Can we please omit the changes to this screen as part of this PR and put them in a separate Pr? :sweat_smile: If it's okay, would like to concentrate on getting the Composables to be high quality and with previews before going furrther.

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        visualTransformation = visualTransformation,
+        trailingIcon = {
+            PasswordVisibility(passwordVisibility = passwordVisibility)
+        },
+        keyboardActions = KeyboardActions.Default,
+    )
+}
+
+@Composable
+fun PasswordVisibility(passwordVisibility: MutableState<Boolean>) {
```
Comment (drawers) [inline app/src/main/java/com/example/materialcomposereader/components/Components.kt:115]: Maybe this Composable can be inlined to the call site? I think it's a little unusual to decompose the problem in this way (with a UI composable called visiblity that really is just an icon)

**Review by drawers — COMMENTED** [review]
Thanks for this! Can we please start with removing the `LoginScreen` changes from this PR and adding Previews for the Composables (if this makes sense)

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt (hunk 1 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt:26]: For now it is only used on LoginScreen, but if implementing a forgot username / password screen it can be reused.

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt (hunk 3 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt:41]: Created issue. Will implement theme typography as ui dev continues.

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt (hunk 7 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt]: resolved

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt (hunk 5 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt]: resolved

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt (hunk 4 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt]: resolved

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt (hunk 2 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt]: resolved

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt (hunk 1 above)
Comment (drawers) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt:28]: Thanks! The reason I asked is because the Compose components in NowInAndroid are a lower layer than this. They seem to be about wrapping a androidx Composable with a themed project Composable. But what you have done here seems okay in this situation if the intention is to reuse between two screens

File: app/src/main/java/com/example/materialcomposereader/components/Components.kt (hunk 7 above)
Comment (drawers) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt]: Thank you for resolving the comments. It's hard to review this PR without previews on the Composables - is it possible to add them in?

Comment (DurtDiv3r) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt]: Sorry, that one got away from me, I'm on it.

Comment (DurtDiv3r) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt]: Resolved. PR updated.

Comment (drawers) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt]: Hmm, I took a look and the previews have been added via a parameterless `@Preview` annotation and by making all the parameters of the Composable itself default. I am not sure this is the correct approach - the default params for a Composable are for the ease of the consumer of those Composables, not for making the preview easier to write. 

I think the best way to do this might be by writing separate private previews in the file. Can you please have a look here 

https://github.com/android/nowinandroid/blob/main/core/ui/src/main/java/com/google/samples/apps/nowinandroid/core/ui/NewsFeed.kt#L119

Comment (DurtDiv3r) [reply app/src/main/java/com/example/materialcomposereader/components/Components.kt]: Resolved in latest commit.

File: app/src/main/java/com/example/materialcomposereader/screens/login/LoginScreen.kt (hunk 6 above)
Comment (drawers) [reply app/src/main/java/com/example/materialcomposereader/screens/login/LoginScreen.kt:19]: Hi @DurtDiv3r  are you able to address this comment please?

**Review by drawers — APPROVED** [review]
